admin/cron/daily.sh

```diff
@@ -94,4 +94,10 @@ fi
 echo `date`" : Updating cover art links"
 ./admin/RebuildCoverArtUrls.pl
 
+echo `date`": Dumping genres.txt.gz"
+./admin/DumpGenresTxt --database MAINTENANCE --output "$TEMP_DIR" --compress
+
+echo `date`": Syncing genres.txt.gz to StaticBrainz"
+./bin/rsync-staticbrainz-files rsync-staticbrainz-mb "$TEMP_DIR"/genres.txt.gz ./
```

Comment:
I was thinking it'd be served from `https://musicbrainz.org/genres.txt` (once an nginx rule is added), and would be documented as an available API.  (So I didn't consider FTP because I didn't think to host an API there.)

Reply:
Which means we can set CORS headers etc. on it, too.

Reply:
If the goal is to serve names through API, I would expect it to be a real-time (cacheable) list rather than a daily snapshot. Here are two potential ways to integrate it into the current MB API:

1. `https://musicbrainz.org/ws/2/genres/all?fmt=txt` with `txt`being a new format that would work only for documented endpoints.
2. `https://musicbrainz.org/ws/2/genres/names` which would ignore any `fmt` parameter.

Reply:
Also relying on an `nginx` rule would be more overhead for deployment/maintenance of prod servers and mirror servers.

Reply:
The `fmt=txt` idea is interesting.  Either of those could work, though it might be odd to introduce a new endpoint that only supports one format, so I might prefer the former.

> I would expect it to be a real-time (cacheable) list rather than a daily snapshot.

Yeah, that wouldn't be wrong -- my thinking was that the genres only change weekly or less, so it would be more efficient to serve a static file for mostly-static content that people are going to poll every time their app starts.  But this might be a case of unneeded optimization.

Reply:
I expect we can still cache the file, at least.

Reply:
@yvanzo do you think `https://musicbrainz.org/ws/2/genres/all?fmt=txt` is better in the long term vs. `https://musicbrainz.org/genres.txt`?  I'm fine with either option, but I'd rather go with what you think is best here.

Reply:
@mwiencek: Yes, it seems to be less complexity, more up-to-date, and still cache-able if needed. The endpoint can also support multiple formats: JSON, XML, TXT.

Maybe we could even more simply have `https://musicbrainz.org/ws/2/genre?fmt=txt` (no plural, no subpath)? (Pattern I suggested in comment to MBS-12093.)

Reply:
I made a typo there: should be genre/all (as it currently is in this PR) not genres/all.  Based on the previous discussion in https://github.com/metabrainz/musicbrainz-server/pull/1298#issuecomment-914042599 I do think the 'all' subpath makes some sense; at present we only use the bare entity path names for searching or browsing, so it might create confusion or at least inconsistency if we have a third type of request without a subpath.

Reply:
Yes it's fine, especially since it's easier to implement and maintain :)
(I updated my comment to the ticket accordingly.)